### PR TITLE
DM-27835 pipetask run should always print stack trace and exception details

### PR DESCRIPTION
**Pull request by n8pease**

### CODE REVIEW

File: python/lsst/daf/butler/tests/cliCmdTestBase.py [hunk 1]
```diff
@@ -50,9 +50,22 @@ def cli(self):
...
+        """Get the mock object to use in place of `mockFunc`. If not provided
+        will use the default provided by `unittest.patch`, this is usually a
+        `unittest.patch.MagicMock`."""
+        return DEFAULT
+
+    @property
+    @abc.abstractmethod
+    def mockFunc(self):
+        """The qualified name of the function to mock, will be passed to
+        unittest.mock.patch, see python docs for details. """
```
Comment (andy-slac) [inline python/lsst/daf/butler/tests/cliCmdTestBase.py:64]: If it is a name and not the function may be better to call it `mockFuncName`?

File: tests/test_cliCmdConfigDump.py [hunk 2]
```diff
@@ -37,6 +37,8 @@
 
 class ConfigDumpTest(CliCmdTestBase, unittest.TestCase):
 
+    mockFunc = "lsst.daf.butler.cli.cmd.commands.script.configDump"
```
Comment (andy-slac) [inline tests/test_cliCmdConfigDump.py]: Base class defines `mockFunc` as abstract property, and this is just an attribute. I guess it should work OK, but if you do not need it to be a property maybe define it as an attribute with `None` value in the base class too?

**Review by andy-slac — APPROVED** [review]
Looks good, couple of minor comments.

**Comment by andy-slac** [comment]
BTW, the `build_and_test` checks are stuck, you need to rebase to master to fix that.

File: tests/test_cliCmdConfigDump.py (hunk 2 above)
Comment (n8pease) [reply tests/test_cliCmdConfigDump.py]: Sadly, it must be a property to work with `@abstract`. But once this is done it satisfies the checker to make it an attribute in the base class. This is the way @timj prefers it, as I understand it.

File: python/lsst/daf/butler/tests/cliCmdTestBase.py (hunk 1 above)
Comment (n8pease) [reply python/lsst/daf/butler/tests/cliCmdTestBase.py:64]: grumbles; yes.

**Pull request merged**